checker.go

```diff
@@ -0,0 +1,229 @@
+// Licensed under the MIT license, see LICENCE file for details.
+
+package quicktest
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"testing"
+	"text/tabwriter"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// ErrorChecker returns a Checker that continues execution in case of failure.
+// The resulting checker checks equality with the given options.
+func ErrorChecker(t *testing.T, opts ...cmp.Option) *Checker {
+	return NewChecker(t.Error, opts...)
+}
+
+// FatalChecker returns a Checker that stops execution in case of failure.
+// The resulting checker checks equality with the given options.
+func FatalChecker(t *testing.T, opts ...cmp.Option) *Checker {
```

Comment:
I'm not entirely convinced how nice this will be to use in practice. I know that my tests often use both Assert and Check in specific places (for example, Assert to assert a non-nil error when we can't do anything useful after it, and Check to check that something else has changed), so in those cases I'd need to create two Checker instances, and still others if I wanted custom equality checking for a particular check. I'd like to see how this actually looks in some real tests - what's the way that these are intended to be used?